### ISSUE TITLE
`carthage update` is checking out the correct version but another version is built

### ISSUE THREAD

**Issue by Vaseltior** [Carthage/Carthage#385]
I had one framework `firstFramework` that contained info about another one `secondFramework`.
I removed all references to the `secondFramework` in my `firstFramework` which means :
- Removing link library with,
- Removing it as a dependency in the cartfile
- Removing it from the tree structure.

I did commit everything, pushed, ok.

    $ carthage update
    *** Fetching xyframeworkswift
    *** Checking out xyframeworkswift at "fd2fbce7e6c3bc54d5e79caeb4745141e96b7764"
    *** xcodebuild output can be found in /var/folders/5n/y5w2hsvs65nchj9vnjxs2jk00000gn/T/carthage-xcodebuild.npc4zA.log
    *** Building scheme "xyframeworkswift" in xyframeworkswift.xcworkspace

Ok great, "fd2fbce7e6c3bc54d5e79caeb4745141e96b7764" is the last commit I did, which actually has all references removed.

But when I look into the Carthage/Build/iOS/xyframeworkswift.framework
I have references to the framework I removed.

Here is my Cartfile

    #
    # XY main framework
    #
    git "git@bitbucket.org:vaseltior/xyframeworkswift.git" "master" 

I tried to remove the ~/Library/Caches/org.carthage.CarthageKit but I had no success.

What cache should I clean? Or is it a bug
Sorry, you're going to hate me

**Comment by jspahrsummers** [comment]
It sounds like you did everything correctly.

> I have references to the framework I removed.

Can you explain what this means in more detail?

**Comment by Vaseltior** [comment]
In the *xyframeworkswift* I added references to *Snap.framework* (Auto layout and constraint framework). I don't use it anymore, so I removed it from the *xyframeworkswift*

Everything seems ok, even the checked out version in the Carthage/Checkouts is ok (no reference to Snap)

Nevertheless, in the Carthage/Build/iOS/xyframeworkswit.framework folder,
I have a Frameworks folder containing Snap.framework.

I also removed every DerivedData I had. I don't know why. I'm going to look at the log file to try to figure out what is going on.
